tests/kani/Invariants/fixme_niche_opt.rs

```diff
@@ -1,38 +0,0 @@
-// Copyright Kani Contributors
-// SPDX-License-Identifier: Apache-2.0 OR MIT
-
-/// This testcase is currently failing with the following error:
-///
-/// [assertion.2] Reached assignment statement with unequal types Pointer { typ:
```

Comment:
Good catch. As we discussed offline, sanity checks that fail should be visible and fail the verification. Thanks!

Reply:
let me address those cases and clean up some of the code as well